source/Ship.h

```diff
@@ -313,21 +308,17 @@ class Ship : public Body, public std::enable_shared_from_this<Ship> {
 	// ship becomes disabled. Returns 0 if the ships hull is already below the
 	// disabled threshold.
 	double HullUntilDisabled() const;
+	// Get this ship's jump navigation, which contains information about how
+	// much it costs for this ship to jump, how far it can jump, and its possible
+	// jump methods.
+	const ShipJumpNavigation &JumpNavigation() const;
+	void RecalibrateJumpNavigation();
 	// Get the number of jumps this ship can make before running out of fuel.
 	// This depends on how much fuel it has and what sort of hyperdrive it uses.
 	// This does not show accurate number of jumps remaining beyond 1.
 	// If followParent is false, this ship will not follow the parent.
 	int JumpsRemaining(bool followParent = true) const;
 	bool NeedsFuel(bool followParent = true) const;
-	// Get the amount of fuel expended per jump.
-	double JumpFuel(const System *destination = nullptr) const;
-	// Get the distance that this ship can jump.
-	double JumpRange(bool getCached = true) const;
-	// Get the cost of making a jump of the given type (if possible).
-	double HyperdriveFuel() const;
-	double JumpDriveFuel(double jumpDistance = 0.) const;
-	std::pair<JumpType, double> GetCheapestJumpType(const System *destination) const;
-	std::pair<JumpType, double> GetCheapestJumpType(const System *from, const System *to) const;
 	// Get the amount of fuel missing for the next jump (smart refuelling)
 	double JumpFuelMissing() const;
```

Comment:
This function could also fit in navigation?

Reply:
JumpFuelMissing

Reply:
JumpFuelMissing requires knowledge of a ship's fuel level and capacity, which JumpNavigation doesn't have.

As stated in the PR description, this is only intended to refactor those functions which can be made more efficient, not to take every possible fuel-related function that could be moved and move it.

Reply:
ok, nevermind then